### DISCUSSION TITLE
Next project builds fine on my local machine, but fails on Vercel

### DISCUSSION THREAD

**Discussion by drew-w** [vercel/next.js#35264] [Help] [answered]
Hey guys, I've been working on a website for a while and finally upgraded my dependancies (including react to "17.0.2" and nextjs to "12.1.1-canary.6"), and updated my next.config to use webpack 5 instead of webpack 4 like i had been using.

Although the project builds just fine on local, when deploying to vercel I get and error:
```

$ next build
info  - Checking validity of types...
info  - Creating an optimized production build...
> Build error occurred
ValidationError: Invalid configuration object. Webpack has been initialized using a configuration object that does not match the API schema.
 - configuration.module.rules[4].issuer has an unknown property 'test'. These properties are valid:
   object { and?, not?, or? }
   -> Logic operators used in a condition matcher.

```

My next.config.js file looks like this:

```
const withImages = require('next-images')
module.exports = withImages({
  env: {
    API_KEY: process.env.API_KEY,
    BASE_URL: process.env.BASE_URL
  },
  images: {
    disableStaticImages: true
  },
  eslint: {
    ignoreDuringBuilds: true
  }
})
```

there is no 'test' property in my webpack config or anywhere in the project for that matter. This is an error that is happening ONLY on Vercel and on local it builds with no issues. This has to be a setting or something on Vercel that is causing this right?

Thanks in advance for your help!

**Comment by icyJoseph** [comment, marked as answer]
Hi,

What version of `next-image` are you using? I found this issue which looks very similar to yours, https://github.com/twopluszero/next-images/issues/42. Try to remove your node_modules locally, and run `npm ci`. You must install one of the latest `next-image` versions where they allowed support for webpack 5.

**Reply by drew-w** [reply]
i was using version 1.4.0, and after updating to the most recent version, it built correctly. 

Any idea why it would build on local using the outdated version?

**Reply by icyJoseph** [reply]
Really hard to tell, perhaps your node version locally is higher than your production environment, or you had installed the latest version, then manually downgraded the version in package.json, but the node_modules still had the latest. Really difficult to tell. 

Usually when I see that production is not matching my local development, I clean the project, install and use `npm explain <package_name>` or `yarn why <package_name>`.

Perhaps you have the dependency unpinned? It may be good to use `x.y.z` over `^x.y.z`.
